src/psyclone/dynamo0p3.py

```diff
@@ -7813,62 +7939,12 @@ def ref_element_properties(self):
-           in self._kern.reference_element.properties:
-            # Query the namespace manager to get the variable name
-            nfaces_h = self._name_space_manager.create_name(
-                root_name="nfaces_re_h", context="PSyVars",
-                label="nfaces_re_h")
-            self._arglist.append(nfaces_h)
-        # Provide no. of vertical faces if required
-        if RefElementMetaData.Property.NORMALS_TO_VERTICAL_FACES \
-           in self._kern.reference_element.properties or \
-           RefElementMetaData.Property.OUTWARD_NORMALS_TO_VERTICAL_FACES \
-           in self._kern.reference_element.properties:
-            nfaces_v = self._name_space_manager.create_name(
-                root_name="nfaces_re_v", context="PSyVars",
-                label="nfaces_re_v")
-            self._arglist.append(nfaces_v)
-        # Now the arrays themselves, in the order specified in the
-        # kernel metadata
-        for prop in self._kern.reference_element.properties:
-            if prop == \
-               RefElementMetaData.Property.OUTWARD_NORMALS_TO_HORIZONTAL_FACES:
-                name = self._name_space_manager.create_name(
-                    root_name="out_normals_to_horiz_faces", context="PSyVars",
-                    label="out_normals_to_horiz_faces")
-                self._arglist.append(name)
-            elif (prop ==
-                  RefElementMetaData.Property.NORMALS_TO_HORIZONTAL_FACES):
-                name = self._name_space_manager.create_name(
-                    root_name="normals_to_horiz_faces", context="PSyVars",
-                    label="normals_to_horiz_faces")
-                self._arglist.append(name)
-            elif (prop == RefElementMetaData.Property.
-                  OUTWARD_NORMALS_TO_VERTICAL_FACES):
-                name = self._name_space_manager.create_name(
-                    root_name="out_normals_to_vert_faces", context="PSyVars",
-                    label="out_normals_to_vert_faces")
-                self._arglist.append(name)
-            elif (prop == RefElementMetaData.Property.
-                  NORMALS_TO_VERTICAL_FACES):
-                name = self._name_space_manager.create_name(
-                    root_name="normals_to_vert_faces", context="PSyVars",
-                    label="normals_to_vert_faces")
-                self._arglist.append(name)
-            else:
-                raise InternalError(
-                    "Unsupported reference-element property ('{0}') found when"
-                    " generating arguments for kernel '{1}'. Supported "
-                    "properties are: {2}".format(
-                        str(prop), self._kern.name,
-                        [str(sprop) for sprop in RefElementMetaData.Property]))
+        # Argument information is produced by a DynReferenceElement
+        # class method
+        if self._kern.reference_element.properties:
+            refelem_args = DynReferenceElement.call_args(self._kern)
```

Comment:
Further to my comment in `call_args`, I don't think it should be (alphabetically) sorting its output as there's no way it can then guarantee the correct ordering of the arguments.

Reply:
As noted in the comments above, the output is not sorted any more.